dd-java-agent/instrumentation/servlet/request-3.1/src/main/java/datadog/trace/instrumentation/servlet31/ServletRequestGetInputStreamInstrumentation.java

```diff
@@ -0,0 +1,130 @@
+package datadog.trace.instrumentation.servlet31;
+
+import static datadog.trace.agent.tooling.ClassLoaderMatcher.hasClassesNamed;
+import static datadog.trace.agent.tooling.bytebuddy.matcher.DDElementMatchers.implementsInterface;
+import static datadog.trace.agent.tooling.bytebuddy.matcher.NameMatchers.named;
+import static datadog.trace.api.gateway.Events.EVENTS;
+import static datadog.trace.bootstrap.instrumentation.api.AgentTracer.activeSpan;
+import static net.bytebuddy.matcher.ElementMatchers.isPublic;
+import static net.bytebuddy.matcher.ElementMatchers.returns;
+import static net.bytebuddy.matcher.ElementMatchers.takesNoArguments;
+
+import com.google.auto.service.AutoService;
+import datadog.trace.agent.tooling.Instrumenter;
+import datadog.trace.api.function.BiFunction;
+import datadog.trace.api.gateway.CallbackProvider;
+import datadog.trace.api.gateway.Flow;
+import datadog.trace.api.gateway.RequestContext;
+import datadog.trace.api.http.StoredBodySupplier;
+import datadog.trace.api.http.StoredByteBody;
+import datadog.trace.bootstrap.instrumentation.api.AgentSpan;
+import datadog.trace.bootstrap.instrumentation.api.AgentTracer;
+import java.nio.charset.Charset;
+import javax.servlet.ServletInputStream;
+import javax.servlet.ServletRequest;
+import javax.servlet.http.HttpServletRequest;
+import net.bytebuddy.asm.Advice;
+import net.bytebuddy.description.type.TypeDescription;
+import net.bytebuddy.matcher.ElementMatcher;
+
+@AutoService(Instrumenter.class)
+public class ServletRequestGetInputStreamInstrumentation extends Instrumenter.AppSec {
+  public ServletRequestGetInputStreamInstrumentation() {
+    super("servlet-request-body");
+  }
+
+  static final ElementMatcher<ClassLoader> CLASS_LOADER_MATCHER =
+      hasClassesNamed("javax.servlet.http.HttpServlet")
+          .and(hasClassesNamed("javax.servlet.ReadListener"));
+
+  @Override
+  public ElementMatcher<ClassLoader> classLoaderMatcher() {
+    return CLASS_LOADER_MATCHER;
+  }
+
+  @Override
+  public ElementMatcher<? super TypeDescription> typeMatcher() {
+    return implementsInterface(named("javax.servlet.ServletRequest"));
+  }
+
+  @Override
+  public void adviceTransformations(AdviceTransformation transformation) {
+    transformation.applyAdvice(
+        named("getInputStream")
+            .and(takesNoArguments())
+            .and(returns(named("javax.servlet.ServletInputStream")))
+            .and(isPublic()),
+        getClass().getName() + "$HttpServletGetInputStreamAdvice");
+  }
+
+  @Override
+  public String[] helperClassNames() {
+    return new String[] {
+      "datadog.trace.instrumentation.servlet31.ServletInputStreamWrapper",
```

Comment:
We usually use the `packageName` helper variable here to make refactors easier
```suggestion
      packageName + ".ServletInputStreamWrapper",
```